server/routers/viewer.tsx

```diff
@@ -392,12 +392,29 @@ const loggedInViewerRouter = createProtectedRouter()
           userId: user.id,
         },
       });
+
       const schedule = availabilityQuery.reduce(
         (schedule: Schedule, availability) => {
           availability.days.forEach((day) => {
             schedule[day].push({
-              start: new Date(new Date().toDateString() + " " + availability.startTime.toTimeString()),
-              end: new Date(new Date().toDateString() + " " + availability.endTime.toTimeString()),
+              start: new Date(
```

Comment:
new Date uses server time, so the previous code breaks on non-UTC server times.

Reply:
Maybe we shouldn't pass these things as `Date`s over the wire at all if the timezone can be an issue?

Maybe this is a better format?


```tsx
type ZeroToNine = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9;
type ZeroToThree = 0 | 1 | 2 | 3;
type ZeroToSix = 0 | 1 | 2 | 3 | 4 | 5 | 6;

type Hour = `0${ZeroToNine}` | `1${ZeroToNine}` | `2${ZeroToThree}`;
type Minute = `${ZeroToSix}${ZeroToNine}`;
type Second = `${ZeroToSix}${ZeroToNine}`;

type TimeObj = {
    hour: Hour,
    monute: Minute,
    second: Second,
    timezone: string;
}
```


Then we can have a function like:

```tsx
function getDate(opts: { time: TimeObj; date?: Date }) {
  const {time, date = new Date()} = opts;
  return dayjs.tz(date, time.timezone ?? ).set("hour", Number(time.hour)).set("minute", Number(time.minute)).set("second", Number(time.second))
}